File: frontend/src/api/detection.ts
import apiClient from './client';

export interface DetectionBox {
  x1: number;
  y1: number;
  x2: number;
  y2: number;
  label: string;
  confidence: number;
}

export interface DetectionResponse {
  id?: number;
  image_url: string;
  detections: DetectionBox[];
  processing_time: number;
}

export interface DetectionHistory {
  id: number;
  image_path: string;
  detection_results: string;
  created_at: string;
}

export const detectionApi = {
  detect: async (file: File): Promise<DetectionResponse> => {
    const formData = new FormData();
    formData.append('file', file);
    
    const response = await apiClient.post('/api/detect', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
    
    return response.data;
  },

  getHistory: async (skip: number = 0, limit: number = 20): Promise<DetectionHistory[]> => {
    const response = await apiClient.get('/api/history', {
      params: { skip, limit },
    });
    return response.data;
  },

  getHistoryDetail: async (id: number): Promise<DetectionHistory> => {
    const response = await apiClient.get(`/api/history/${id}`);
    return response.data;
  },

  deleteHistory: async (id: number): Promise<void> => {
    await apiClient.delete(`/api/history/${id}`);
  },
};

File: backend/app/schemas/detection.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime


class DetectionBox(BaseModel):
    x1: float
    y1: float
    x2: float
    y2: float
    label: str
    confidence: float


class DetectionRequest(BaseModel):
    pass  # 画像はファイルアップロードで送信


class DetectionResponse(BaseModel):
    id: Optional[int] = None
    image_url: str
    detections: List[DetectionBox]
    processing_time: float


class DetectionHistoryResponse(BaseModel):
    id: int
    image_path: str
    detection_results: str
    created_at: datetime

    class Config:
        from_attributes = True
